src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/CrossTarget/AggregateCrossTargetProjectContext.cs

```diff
@@ -7,78 +7,49 @@
-            _activeTargetFramework = activeTargetFramework;
-            TargetFrameworkProvider = targetFrameworkProvider;
+            ActiveTargetFramework = activeTargetFramework;
+            _targetFrameworkProvider = targetFrameworkProvider;
         }
 
-        private ITargetFrameworkProvider TargetFrameworkProvider { get; }
-
-        public IEnumerable<ITargetedProjectContext> InnerProjectContexts => _configuredProjectContextsByTargetFramework.Values;
-
         public IEnumerable<ConfiguredProject> InnerConfiguredProjects => _configuredProjectsByTargetFramework.Values;
 
-        public ITargetedProjectContext ActiveProjectContext => _configuredProjectContextsByTargetFramework[_activeTargetFramework];
-
-        public bool IsCrossTargeting { get; }
-
-        public void SetProjectFilePathAndDisplayName(string projectFilePath, string displayName)
-        {
-            // Update the project file path and display name for all the inner project contexts.
-            foreach ((ITargetFramework targetFramework, ITargetedProjectContext innerProjectContext) in _configuredProjectContextsByTargetFramework)
-            {
-                // For cross targeting projects, we ensure that the display name is unique per every target framework.
-                innerProjectContext.DisplayName = IsCrossTargeting ? $"{displayName}({targetFramework})" : displayName;
-                innerProjectContext.ProjectFilePath = projectFilePath;
-            }
-        }
-
-        public ITargetedProjectContext GetInnerProjectContext(ProjectConfiguration projectConfiguration, out bool isActiveConfiguration)
+        public ITargetFramework GetProjectFramework(ProjectConfiguration projectConfiguration)
         {
             if (projectConfiguration.IsCrossTargeting())
             {
-                string targetFrameworkMoniker =
-                    projectConfiguration.Dimensions[ConfigurationGeneral.TargetFrameworkProperty];
-                ITargetFramework targetFramework = TargetFrameworkProvider.GetTargetFramework(targetFrameworkMoniker);
+                string targetFrameworkMoniker = projectConfiguration.Dimensions[ConfigurationGeneral.TargetFrameworkProperty];
 
-                isActiveConfiguration = _activeTargetFramework.Equals(targetFramework);
-
-                return _configuredProjectContextsByTargetFramework.TryGetValue(targetFramework, out ITargetedProjectContext projectContext)
-                    ? projectContext
-                    : null;
+                return _targetFrameworkProvider.GetTargetFramework(targetFrameworkMoniker);
             }
             else
             {
-                isActiveConfiguration = true;
-                if (_configuredProjectContextsByTargetFramework.Count > 1)
-                {
-                    return null;
-                }
-
-                return InnerProjectContexts.Single();
+                return TargetFrameworks.Length > 1 ? null : TargetFrameworks[0];
```

Comment:
Is the length check necessary?

Reply:
Maybe not. `IsCrossTargeting` returns true if the project configuration contains a dimension for `TargetFrameworkProperty`. This check was present in the prior code and I'm not sure it's safe to remove just yet.

At some point soon all this cross targeting stuff will be removed in favour of a purely dimension driven approach where a question like this should have a self evident answer. At least, that's the theory.